src/imitation/data/fragments.py

```diff
@@ -0,0 +1,87 @@
+import random
+import warnings
+from typing import Callable, List, Optional, Tuple
+
+from imitation.data.types import TrajectoryWithRew
+
+Fragmenter = Callable[
+    [List[TrajectoryWithRew]], List[Tuple[TrajectoryWithRew, TrajectoryWithRew]]
+]
+"""Creates pairs of trajectory fragments from a collection of trajectories."""
+
+
+class RandomFragmenter:
+    """Sample fragments of trajectories uniformly at random with replacement.
+
+    Note that each fragment is part of a single episode and has a fixed
+    length. This leads to a bias: transitions at the beginning and at the
+    end of episodes are less likely to occur as part of fragments (this affects
+    the first and last fragment_length transitions).
+
+    TODO(ejnnr): should we correct for this bias and make all transitions equally
+    likely to be part of fragments?
+    """
+
+    def __init__(
+        self,
+        fragment_length: int = 50,
+        num_pairs: int = 50,
+        seed: Optional[float] = None,
+    ):
+        """Initialize the fragmenter.
+
+        Args:
+            fragment_length: the length of each sampled fragment
+            num_pairs: the number of fragment pairs to sample
+            seed: an optional seed for the internal RNG
+        """
+        self.fragment_length = fragment_length
+        self.num_pairs = num_pairs
+        self.rng = random.Random(seed)
+
+    def __call__(
+        self, trajectories: List[TrajectoryWithRew]
+    ) -> List[Tuple[TrajectoryWithRew, TrajectoryWithRew]]:
+        fragments: List[TrajectoryWithRew] = []
+
+        # filter out all trajectories that are too short
+        trajectories = [
+            traj for traj in trajectories if len(traj) >= self.fragment_length
+        ]
+        if len(trajectories) == 0:
+            raise ValueError(
+                "No trajectories are long enough for the desired fragment length."
+            )
+
+        weights = [len(traj) for traj in trajectories]
+
+        # TODO(ejnnr): since we're sampling with replacement, there could
+        # already be lots of duplicates below this threshold. Perhaps
+        # we should warn already if the number of transitions isn't larger
```

Comment:
Bumping warning threshold seems good to me.

Sampling without replacement is probably not worth the hassle. In most applications of DRLHP, human evaluation is expensive, collecting samples is cheap. So we'd expect `sum(weights)` to be much larger than `self.num_pairs`.